### PR TITLE
LogbookAutoConfiguration

### PR DESCRIPTION
**Pull request by whiskeysierra**

### CODE REVIEW

**Comment by whiskeysierra** [comment]
Is still missing documentation and **a lot** of tests.

**Comment by whiskeysierra** [comment]
This fixes #25

File: README.md [hunk 1]
```diff
@@ -292,7 +292,7 @@ You may obtain a copy of the License at
 
     http://www.apache.org/licenses/LICENSE-2.0
 
-Unless required by applicable law or agreed to in writing, software
+Unless required by applicable law or agreed to in write, software
```
Comment (jhorstmann) [inline README.md]: Refactoring gone wrong?

Comment (whiskeysierra) [inline README.md]: > Refactoring gone wrong?

Indeed

File: logbook-core/pom.xml [hunk 2]
```diff
@@ -34,7 +34,6 @@
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-surefire-plugin</artifactId>
                 <configuration>
```
Comment (lukasniemeier-zalando) [inline logbook-core/pom.xml]: ?

File: logbook-spring-boot-starter/src/main/java/org/zalando/logbook/spring/LogbookAutoConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+    @ConditionalOnMissingBean(HttpLogFormatter.class)
+    @ConditionalOnProperty(name = "logbook.format.style", havingValue = "http")
+    public HttpLogFormatter httpFormatter() {
+        return new DefaultHttpLogFormatter();
+    }
+
+    @Bean
+    @ConditionalOnBean(ObjectMapper.class)
+    @ConditionalOnMissingBean(HttpLogFormatter.class)
+    public HttpLogFormatter jsonFormatter(
```
Comment (lukasniemeier-zalando) [inline logbook-spring-boot-starter/src/main/java/org/zalando/logbook/spring/LogbookAutoConfiguration.java]: What happens if you have Jackson on the classpath and style set to `http` ?

Comment (whiskeysierra) [inline logbook-spring-boot-starter/src/main/java/org/zalando/logbook/spring/LogbookAutoConfiguration.java]: > What happens if you have Jackson on the classpath and style set to http ?

I checked and my current test is configuring the objectmapper in both tests, but as son as we specify http style, we get the http formatter.

File: logbook-core/pom.xml (hunk 2 above)
Comment (whiskeysierra) [inline logbook-core/pom.xml]: > ?

Cleaned up

**Pull request merged**